src/TSID/include/BipedalLocomotion/TSID/OptimalControlElement.h

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file OptimalControlElement.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_TSID_OPTIMAL_CONTROL_ELEMENT_H
+#define BIPEDAL_LOCOMOTION_TSID_OPTIMAL_CONTROL_ELEMENT_H
+
+#include <memory>
+
+#include <Eigen/Dense>
+
+#include <iDynTree/KinDynComputations.h>
+
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/VariablesHandler.h>
+
+namespace BipedalLocomotion
+{
+namespace TSID
+{
+
+/**
+ * OptimalControlElement describes a control problem element. The element is described by a matrix
+ * \f$A\f$ and a vector \f$b\f$. This class describes both a linear equality constraint and a linear
+ * inequality constraint. In case of equality constraint \f$ A \f$ and \f$ b \f$ represents: \f$ Ax
+ * = b\f$ In case of inequality constraint \f$ A \f$ and \f$ b \f$ represents: \f$ Ax \le b \f$
+ * @note Please inherit this class if you want to build your own optimal control problem.
+ */
+class OptimalControlElement
+{
+protected:
+    Eigen::MatrixXd m_A; /**< Element Matrix */
+    Eigen::VectorXd m_b; /**< Element Vector */
```

Comment:
I would suggest to name this in a way that is more specific to TSID, like ``TaskElement``, hiding how that TSID problem is solved (either via optimal control, pseudoinverse, or other methods). ``OptimalControl`` is, in my opinion, generic and a bit misleading, considering that you are considering only linear elements.

Reply:
here: https://github.com/dic-iit/bipedal-locomotion-framework/pull/167/commits/f7cd662288b4c945ea3dbc8be285968027117feb I renamed the class as `Task`